.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,36 @@
+{
+  "name": "oauth2-proxy Dev",
+  "context": "..",
+  "dockerFile": "../Dockerfile.devcontainer",
+  "postCreateCommand": "pre-commit install && go mod download",
+  "containerEnv": { "DEVCONTAINER": "1" },
+  "appPort": ["4180:4180"],
+  "runArgs": ["-e", "GIT_EDITOR=code --wait"],
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "golang.go",
+        "ms-vscode.makefile-tools",
+        "ms-azuretools.vscode-docker",
+        "visualstudioexptteam.vscodeintellicode",
+        "redhat.vscode-yaml",
+        "esbenp.prettier-vscode",
+        "GitHub.vscode-pull-request-github"
+      ],
```

Comment:
Furthermore, I like to add the features:

https://github.com/devcontainers/features/tree/main/src/docker-in-docker
https://github.com/devcontainers/features/tree/main/src/kubectl-helm-minikube
https://github.com/devcontainers/features/tree/main/src/common-utils 
and enable configureZshAsDefaultShell

Reply:
Done